### PR TITLE
[BugFix] Fix includeTypeName status is incorrect When the es version …

### PR DESCRIPTION
**Pull request by kobebryantlin0**

### CODE REVIEW

**Comment by kobebryantlin0** [comment]
您好，我是林恩伟，你发的邮件我已经收到。

**Comment by Seaven** [comment]
run starrocks_fe_unittest

File: fe/fe-core/src/main/java/com/starrocks/external/elasticsearch/MappingPhase.java [hunk 1]
```diff
@@ -49,6 +49,8 @@ public MappingPhase(EsRestClient client) {
     public void preProcess(SearchContext context) {
         if (context.version() != null && context.version().onOrAfter(EsMajorVersion.V_7_X)) {
             includeTypeName = true;
+        } else if (context.version() != null && context.version().before(EsMajorVersion.V_7_X)) {
+            includeTypeName = false;
```
Comment (RowenWood) [inline fe/fe-core/src/main/java/com/starrocks/external/elasticsearch/MappingPhase.java:53]: includeTypeName  default value is false， why add this line？

Comment (kobebryantlin0) [reply fe/fe-core/src/main/java/com/starrocks/external/elasticsearch/MappingPhase.java:53]: When the es is down or crashed, the SR loses the connection with the ES , and the version is set to 7.x(EsMajorVersion.LATEST) in VersionPhase#execute function by default, and then the field is set to True (MappingPhase # preProcess). After the es is restarted and restored, the version number is restored to 6. x, but the includeTypeName is still true, not false.

I have emailed seaven to describe our specific scenario.

**Comment by Seaven** [comment]
run starrocks_clang-format starrocks_clang-tidy starrocks_tscancode

**Comment by Smith-Cruise** [comment]
You should request pr for main branch first, I think this problem still existed in main branch.

**Review by Smith-Cruise — COMMENTED** [review]
LGTM

File: fe/fe-core/src/main/java/com/starrocks/external/elasticsearch/MappingPhase.java (hunk 1 above)
Comment (kobebryantlin0) [reply fe/fe-core/src/main/java/com/starrocks/external/elasticsearch/MappingPhase.java:53]: @RowenWood

**Comment by kobebryantlin0** [comment]
> You should request pr for main branch first, I think this problem still existed in main branch.

i

> LGTM

@Smith-Cruise the main branch made a huge reconstruction after branch-2.3, and i find the function(MappingPhase#preProcess) has been removed,  i am not familar with those change now. could you review and merge to version 2.3  first？

I also have emailed you to describe our specific scenario.

**Review by Seaven — APPROVED** [review]

**Comment by Seaven** [comment]
@Smith-Cruise Follow up this BugFix on main&2.2&2.4&2.5 plz

**Comment by Smith-Cruise** [comment]
> > You should request pr for main branch first, I think this problem still existed in main branch.
> 
> i
> 
> > LGTM
> 
> @Smith-Cruise the main branch made a huge reconstruction after branch-2.3, and i find the function(MappingPhase#preProcess) has been removed, i am not familar with those change now. could you review and merge to version 2.3 first？
> 
> I also have emailed you to describe our specific scenario.

You can bug-fix in 2.4 branch, then cherry-pick to 2.3 & 2.2

**Pull request merged**